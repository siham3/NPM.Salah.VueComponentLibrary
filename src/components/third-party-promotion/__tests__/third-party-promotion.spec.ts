import { shallowMount, mount } from "@vue/test-utils";
import { expect, it, describe } from "vitest";
import ThirdPartyPromotion from "../third-party-promotion.vue";
import type { ThirdPartyPromotionProps } from "../types";

const defaultProps: ThirdPartyPromotionProps = {
	iconUrl: "/images/iconurl.svg",
	content: "My Content",
};

interface RenderOptions {
	props: ThirdPartyPromotionProps;
}

const buildRenderOptions = (
	props: ThirdPartyPromotionProps
): RenderOptions => ({ props });

describe("third-party-promotion default variant", () => {
	it("matches snapshot", () => {
		const wrapper = shallowMount(
			ThirdPartyPromotion,
			buildRenderOptions(defaultProps)
		);
		expect(wrapper.element).toMatchSnapshot();
	});

	it("emits button-click", async () => {
		const buttonText = "This is button text";
		const wrapper = mount(
			ThirdPartyPromotion,
			buildRenderOptions({ ...defaultProps, buttonText })
		);

		const button = wrapper.find("button");
		await button.trigger("click");

		expect(button.text()).toBe(buttonText);
		expect(wrapper.emitted()).toHaveProperty("buttonClick");
	});
});
