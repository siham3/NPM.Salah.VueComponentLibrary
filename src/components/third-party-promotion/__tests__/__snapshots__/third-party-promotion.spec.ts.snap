// Vitest Snapshot v1

exports[`third-party-promotion default variant > matches snapshot 1`] = `
<div
  class="third-party-promotion"
  data-v-2c558045=""
>
  <div
    class="icon"
    data-v-2c558045=""
  >
    <img
      data-v-2c558045=""
      src="/images/iconurl.svg"
    />
  </div>
  <sequiter-markdown-stub
    class="content"
    content="My Content"
    data-v-2c558045=""
  />
  <div
    class="button-bar"
    data-v-2c558045=""
  >
    <sequiter-button-stub
      data-v-2c558045=""
      disabled="false"
      larger="false"
      type="button"
      variant="primary"
    />
  </div>
</div>
`;
